src/EditorFeatures/Core.Wpf/Classification/ClassificationTypeFormatDefinitions.cs

```diff
@@ -372,7 +372,171 @@ private XmlDocCommentTextFormatDefinition()
+        }
+
+        [Export(typeof(EditorFormatDefinition))]
+        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RegexGrouping)]
+        [Name(ClassificationTypeNames.RegexGrouping)]
+        [Order(After = ClassificationTypeNames.StringLiteral)]
+        [Order(After = ClassificationTypeNames.VerbatimStringLiteral)]
+        [UserVisible(true)]
+        [ExcludeFromCodeCoverage]
+        private class RegexGroupingFormatDefinition : ClassificationFormatDefinition
+        {
+            private RegexGroupingFormatDefinition()
+            {
+                this.DisplayName = EditorFeaturesWpfResources.Regex_Grouping;
+                this.ForegroundColor = s_regexGroupingAndAlternationColor;
+            }
+        }
+
+        [Export(typeof(EditorFormatDefinition))]
+        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RegexAlternation)]
+        [Name(ClassificationTypeNames.RegexAlternation)]
+        [Order(After = ClassificationTypeNames.StringLiteral)]
+        [Order(After = ClassificationTypeNames.VerbatimStringLiteral)]
+        [UserVisible(true)]
+        [ExcludeFromCodeCoverage]
+        private class RegexAlternationFormatDefinition : ClassificationFormatDefinition
+        {
+            private RegexAlternationFormatDefinition()
+            {
+                this.DisplayName = EditorFeaturesWpfResources.Regex_Alternation;
+                this.ForegroundColor = s_regexGroupingAndAlternationColor;
+            }
+        }
+
+        [Export(typeof(EditorFormatDefinition))]
+        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RegexText)]
+        [Name(ClassificationTypeNames.RegexText)]
+        [Order(After = ClassificationTypeNames.StringLiteral)]
+        [Order(After = ClassificationTypeNames.VerbatimStringLiteral)]
+        [UserVisible(true)]
+        [ExcludeFromCodeCoverage]
+        private class RegexTextFormatDefinition : ClassificationFormatDefinition
+        {
+            private RegexTextFormatDefinition()
+            {
+                this.DisplayName = EditorFeaturesWpfResources.Regex_Text;
+                this.ForegroundColor = s_regexTextColor;
+            }
+        }
+
+        [Export(typeof(EditorFormatDefinition))]
+        [ClassificationType(ClassificationTypeNames = ClassificationTypeNames.RegexSelfEscapedCharacter)]
+        [Name(ClassificationTypeNames.RegexSelfEscapedCharacter)]
+        [Order(After = ClassificationTypeNames.StringLiteral)]
+        [Order(After = ClassificationTypeNames.VerbatimStringLiteral)]
+        [UserVisible(true)]
+        [ExcludeFromCodeCoverage]
+        private class RegexSelfEscapedCharacterFormatDefinition : ClassificationFormatDefinition
+        {
+            private RegexSelfEscapedCharacterFormatDefinition()
+            {
+                this.DisplayName = EditorFeaturesWpfResources.Regex_SelfEscapedCharacter;
+                this.ForegroundColor = s_regexTextColor;
```

Comment:
Maybe a comment as to why this uses the `Text` color? Or rename the field to account for both?

Reply:
done.